implemented.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"strings"
+)
+
+// implementedFuncs returns list of Func which already implemented.
+func implementedFuncs(fns []Func, recv string, srcDir string) (map[string]bool, error) {
+
+	// determine name of receiver type
+	recvType := getReceiverType(recv)
+
+	fset := token.NewFileSet()
+	pkgs, err := parser.ParseDir(fset, srcDir, nil, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	implemented := make(map[string]bool)
+
+	// getReceiver returns title of struct to which belongs the method
+	getReceiver := func(mf *ast.FuncDecl) string {
+		if mf.Recv == nil {
+			return ""
+		}
+
+		for _, v := range mf.Recv.List {
+			switch xv := v.Type.(type) {
+			case *ast.StarExpr:
+				if si, ok := xv.X.(*ast.Ident); ok {
+					return si.Name
+				}
+			case *ast.Ident:
+				return xv.Name
+			}
+		}
+
+		return ""
+	}
+
+	// finder is a walker func which will be called for each element in the source code of package
+	// but we are interested in funcs only with receiver same to typeTitle
+	finder := func(n ast.Node) bool {
+		x, ok := n.(*ast.FuncDecl)
+		if !ok {
+			return true
+		}
+		if getReceiver(x) != recvType {
+			return true
+		}
+		for _, fn := range fns {
+			if fn.Name == x.Name.String() {
+				implemented[fn.Name] = true
+				break
```

Comment:
On a second read, I think we should remove this break.

Without the break, we build a list of all methods declared.

With the break, we do O(num methods requested X num methods declared) work, which is quadratic.

The number of methods requested is likely to be small in almost all cases, but I can imagine it blowing up painfully in large, autogenerated files. My years of working on the compiler has taught me that people do weird and wonderful things. :)

What do you think?

Reply:
Hm, but what will it give us? This cycle only search for first method which already implemented and when we've found it we save it to map. Here the logic is finish. But what do you propose it should do after it founds method? I don't understand why we should continue comparison after our job is finished even if we have large autogenerated file?

Reply:
Sorry, I did a bad job explaining.

But here's a simpler idea: Convert fns from a []string to a map[string]bool. Then do a map lookup here instead of a loop. This will avoid the O(n) loop that could in theory create quadratic behavior.

Reply:
fns is not a []string. fns is []Func

Reply:
I'll make the change in a follow-up. Thanks again for the contribution! This is a much-requested feature.